pkgsrc/p4-examples/README.md

```diff
@@ -0,0 +1,79 @@
+P4 Examples Package
+===================
+
+P4_16 programs and PTF test scripts corresponding to each of them.
+    
+The `p4-examples` package is organized as follows
+* `p4_16_programs`
+Example P4_16 programs
+* `programs`
+Example P4_14 programs/to    
+* `ptf-tests`
+PTF tests corresponding to each of the example P4_14 programs in the `programs` directory
+    Makefile:
+        Makefile to install conf files needed by switchd application for
+        PTF tests
+
+Building and installing
+=======================
+
+Building and installation of the examples is done using the `p4studio` tool. It is possible to either build the examples during the `open-p4studio` installation or to build them later if desired. The latter option saves time during the initial installation and does eliminate possible issues with the examples that can occasionally become broken. 
+
+It is possible to build all the examples, just P4_14 or P4_16 programs as well as each individual one. In all cases we assume that `open-p4studio` has already been installed and the critical environment variables have been set either by *sourcing* the script `~/setup-open-p4studio.bash`, manually or otherwise.
+
+To build an individual example (be it a P4_14 or a P4_16 one) use the command below. Each example is located in its separate directory and the name of that directory is the example's name. The reason we use this terminology (i.e., "example_name" instead of "program_name" is because some examples (e.g. `tna_32q_multiprogram`) might consist of more than one P4 program:
+
+```
+$SDE/p4studio/p4studio build <example_name> 
+```
+
+To build all P4_16 examples use the command:
+
+```
+$SDE/p4studio/p4studio build p4-16-programs
+```
+
+To build all P4_14 examples use the command:
+
+```
+$SDE/p4studio/p4studio build p4-16-programs
```

Comment:
Shouldn't the command to build all P4_14 programs be different than the one to build all P4_16 programs above?